libsolutil/CDCL.h

```diff
@@ -113,7 +237,43 @@ class CDCL
 	std::vector<size_t> m_decisionPoints;
 	/// Index into assignmentTrail: All assignments starting there have not yet been propagated.
 	size_t m_assignmentQueuePointer = 0;
+
+	// Current state of the solver. If FALSE, we are in an UNSAT state.
+	bool ok = true;
+
+	// Proof log
+	std::ostream* proof = nullptr;
+	uint64_t clause_ID = 1;
+	std::map<Literal, uint64_t> unit_cl_IDs;
+	uint64_t unsat_clause_ID = 0;
+	void writeFinalProofClauses();
 };
 
+inline bool CDCL::isAssigned(Literal const& _literal) const
+{
+	return m_assignments[_literal.variable] != tristate_unset;
+}
+
+inline bool CDCL::isAssignedTrue(Literal const& _literal) const
+{
+	return isAssigned(_literal) &&
+		(m_assignments[_literal.variable].toBool() ^ !_literal.positive) == true;
```

Comment:
Isn't this just
```suggestion
	return m_assignments[_literal.variable] == _literal.positive;
```
now? If yes, do we actually still need the helper?

Reply:
The way this is done in most SAT solvers, is to have an inline function `TriState CDCL:value(const Literal& x)` that returns a `TriState`. So then things get nice in the code, because you can call it like:

```
if (value(cl[0]) == tristate_true) {
  // blah
};
```

Also, `value` is then defined for variables, too, so you don't have to care if you put in a variable or a literal. Obviously, for variables, it just returns `m_assignments[var]` :smiley:

Reply:
Indeed, sounds good!

Reply:
Added this as part of the `constexpr static` system for `TriState`. We only use `m_assignments` when we need to assign a variable a value, i.e. only inside `enqueue()` and `cancelUntil()`. This makes the code easier to review, in my opinion, as value() cannot change the assignment of a variable.